juntagrico_assignment_request/entity/assignment_request.py

```diff
@@ -125,11 +138,13 @@ def get_matching_job(self):
     @classmethod
     def _create_or_update_assignment(cls, instance):
         instance.set_activityarea_if_none()
+
         # assignment
         if instance.assignment:
             # update if exists:
             # saving the assignment will automatically restore its content. If old job is empty after that, remove it.
-            cls._remove_job(instance.assignment.job, instance.assignment.save)
+            instance.assignment.save()
+            cls._remove_job(instance.assignment.job)
 
```

Comment:
We don't have to re-apply the amounts here because they are already set when assignments are saved in `pre_save_assignment`